modules/unsupported/solr/src/main/java/org/geotools/data/solr/SolrFeatureReader.java

```diff
@@ -168,7 +170,9 @@ public SimpleFeature next() throws IOException, IllegalArgumentException,
                 if (type instanceof GeometryDescriptor) {
                     GeometryDescriptor gatt = (GeometryDescriptor) type;
                     if (value != null) {
-                        Geometry geometry = wktReader.read(value.toString());
+                        SolrSpatialStrategy spatialStrategy = SolrSpatialStrategy.createStrategy(gatt);
+                        Geometry geometry = spatialStrategy.decode(value.toString());

```

Comment:
Wondering if creating the strategy object at each feature read might be too much overhead.. maybe it should be created once when the reader is instantiated instead?